### ISSUE TITLE
[TODO_BLOCKER] Red0ne's Iteration16 TODOs

### ISSUE THREAD

**Issue by red-0ne** [pokt-network/poktroll#572]
## Objective

Tackle all on-chain blockers assigned to red-0ne

## Origin Document

- https://github.com/pokt-network/poktroll/pull/577
![Screenshot 2024-06-03 at 10 09 12 PM](https://github.com/pokt-network/poktroll/assets/1892194/3ccd2cbf-a09c-433a-9aa9-aff46004be05)

## Goals

- Tackle all major TODO's in the codebase assigned to red-0ne

## Deliverables

The output of the following command from https://github.com/pokt-network/poktroll/pull/577:

```bash
grep -r "TODO_BLOCKER(@red-0ne)" . | sed 's/^/- [ ] `/' | sed 's/:/`:/'
```

- [x] `./x/tokenomics/keeper/keeper_test.go`:   // TODO_BLOCKER(@red-0ne): Use the governance parameters for more
- [x] `./x/tokenomics/keeper/keeper_test.go`:   // TODO_BLOCKER(@red-0ne): Use the governance parameters for more precise block heights once they are implemented.
- [x] `./x/tokenomics/keeper/keeper_test.go`:   // TODO_BLOCKER(@red-0ne): Use the governance parameters for more precise block heights once they are implemented.
- [x] `./x/tokenomics/keeper/keeper_test.go`:   // TODO_BLOCKER(@red-0ne): Use the governance parameters for more precise block heights once they are implemented.
- [x] `./x/supplier/config/supplier_configs_reader.go`: // TODO_BLOCKER(@red-0ne): This hasn't been implemented yet but is part of the
- [x] `./x/proof/keeper/msg_server_submit_proof.go`:    // TODO_BLOCKER(@red-0ne): Fetch the correct ring for the session this relay is from.
- [x] `./pkg/partials/payloads/rest.go`:        // TODO_BLOCKER(@red-0ne): Implement this function
- [x] `./pkg/partials/payloads/rest.go`:        // TODO_BLOCKER(@red-0ne): Implement this function
- [x] `./pkg/partials/payloads/rest.go`:        // TODO_BLOCKER(@red-0ne): Implement this method
- [x] `./pkg/partials/payloads/rest.go`:        // TODO_BLOCKER(@red-0ne): Implement this method
- [x] `./pkg/partials/payloads/jsonrpc.go`:     // TODO_BLOCKER(@red-0ne): Implement this method
- [x] `./pkg/partials/partial.go`:// TODO_BLOCKER(@red-0ne): This function currently only supports JSON-RPC and must
- [x] `./pkg/partials/partial.go`:      // TODO_BLOCKER(@red-0ne): Handle other request types
- [x] `./testutil/testproxy/relayerproxy.go`:// TODO_BLOCKER(@red-0ne): This function only supports JSON-RPC requests and
- [ ] `./e2e/tests/session.feature`:# TODO_BLOCKER(@red-0ne): Make sure to implement and validate this test

---

**Creator**: @red-0ne

**Comment by Olshansk** [comment]
@red-0ne I recommend running `grep -r "TODO_BLOCKER(@red-0ne )" . | sed 's/^/- [ ] `/' | sed 's/:/`:/'` to update the deliervables after #577 is merged in.

**Comment by red-0ne** [comment]
Closed as per #694
